packages/flutter_tools/templates/create/android.tmpl/app/src/main/java/com/yourcompany/projectName/MainActivity.java.tmpl

```diff
@@ -2,16 +2,15 @@ package {{androidIdentifier}};
 
 import android.os.Bundle;
 import io.flutter.app.FlutterActivity;
-{{#withPluginHook}}
-import {{androidPluginIdentifier}}.{{pluginClass}};
-{{/withPluginHook}}
+import io.flutter.plugins.PluginRegistry;
 
 public class MainActivity extends FlutterActivity {
+    PluginRegistry pluginRegistry;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-{{#withPluginHook}}
-        {{pluginClass}}.register(this);
-{{/withPluginHook}}
+        pluginRegistry = new PluginRegistry();
+        pluginRegistry.registerAll(this);
```

Comment:
If we had the plugin registry code happen in FlutterActivity's onCreate we could do this automatically for all created FlutterActivities. It would be more robust to developers accidentally deleting the auto-registration code, or the registry API changing and breaking client apps.

Another possibility that's maybe better long-term is for the registry to be owned by the FlutterApplication with lifecycle callbacks rather than having registration code happen in Activity onCreate. This is useful for plugins that want to do something when the application is running but has no visible activities (e.g. push messaging). It's fine if this is beyond the scope of what you're trying to do with change, but the nice thing about moving plugin registration out of the client code is that we'd be able to do it invisibly without breaking apps.

/cc @goderbauer